packages/models-tree/src/models-tree/ModelsTreeDefinition.ts

```diff
@@ -559,6 +564,33 @@ class SubjectModelIdsCache {
       });
   }
 
+  public async getParentSubjectIds(): Promise<Id64String[]> {
+    await this.initCache();
+    if (!this._parentSubjectIds) {
+      this._parentSubjectIds = (async () => {
```

Comment:
`this._parentSubjectIds` doesn't have to be a promise. Also I think you can apply a `??=` operator here.

Reply:
> `this._parentSubjectIds` doesn't have to be a promise.

why?

Reply:
There are no asynchronous operations needed to create it, so it can be stored as an array instead of a promise. Am I missing something?

Reply:
Gotcha. The `initCache` should've been inside the parent subjects function. Made the change together with the suggested use of `??=` operator.